Services/Mail/classes/AutoResponder/class.ilAutoResponderServiceImpl.php

```diff
@@ -0,0 +1,117 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ ********************************************************************
+ */
+
+class ilAutoResponderServiceImpl implements ilAutoResponderService
+{
+    protected bool $auto_responder_status;
+    protected array $auto_responder_data;
+    protected int $global_idle_time_interval;
+    /** @var callable  */
+    protected $loginByUsrIdCallable;
+    protected ilAutoResponderRepository $auto_responder_repository;
+
+    public function __construct(
+        callable $loginByUsrIdCallable = null,
+        int $global_idle_time_interval = null,
+        bool $auto_responder_status = false,
+        array $auto_responder_data = [],
+        ilAutoResponderRepository $auto_responder_repository = null
+    ) {
+        global $DIC;
+        $this->global_idle_time_interval = $global_idle_time_interval ?? (int) $DIC->settings()->get('mail_auto_responder_idle_time');
+        $this->loginByUsrIdCallable = $loginByUsrIdCallable ?? static function (int $usrId) : string {
+            return ilObjUser::_lookupLogin($usrId);
+        };
+        $this->auto_responder_status = $auto_responder_status;
+        $this->auto_responder_data = $auto_responder_data;
+        $this->auto_responder_repository = $auto_responder_repository ?? new ilAutoResponderDatabaseRepository();
+    }
+
+    public function isAutoResponderEnabled() : bool
+    {
+        return $this->auto_responder_status;
+    }
+
+    public function enableAutoResponder() : void
+    {
+        $this->auto_responder_status = true;
+    }
+
+    public function disableAutoResponder() : void
+    {
+        $this->auto_responder_status = false;
+    }
+
+    public function handleAutoResponderMails(int $receiver_usr_id) : void
+    {
+        if ($this->auto_responder_data) {
+            foreach ($this->auto_responder_data as $usr_id => $mail_options) {
+                $auto_responder = $this->auto_responder_repository->findBySenderIdAndReceiverId(
+                    $usr_id,
+                    $receiver_usr_id
+                );
+                if (!$auto_responder) {
+                    $auto_responder = new ilAutoResponder(
+                        $usr_id,
+                        $receiver_usr_id,
+                        (new DateTimeImmutable('NOW'))->sub(new DateInterval('P' . $this->global_idle_time_interval . 'D'))
```

Comment:
Please use the `Clock` data type available in `src/Data` to receive `now`. This factory should be a dependency of this service and injected in the constructor.